src/os/unix.rs

```diff
@@ -251,23 +249,23 @@ fn getpwuid(name: Name) -> Result<OsString> {
             target_os = "openbsd",
         ))]
         {
+            let result: *mut *mut PassWd = null_mut();
             let ret = getpwuid_r(
                 geteuid(),
                 passwd.as_mut_ptr(),
                 buffer.as_mut_ptr() as *mut c_void,
                 BUF_SIZE,
-                _passwd.as_mut_ptr(),
+                result,
             );
 
             if ret != 0 {
                 return Err(Error::last_os_error());
             }
 
-            let _passwd = _passwd.assume_init();
-
-            if _passwd.is_null() {
+            if result.is_null() {
```

Comment:
This would always return `true`, meaning this function can't succeed.

I also don't believe it's guaranteed that the `result` argument is allowed to be `NULL`.  The argument needs to be a pointer to a pointer, where we can check if it the overwritten pointer is null.

Reply:
I dropped this commit.